plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteTypeMapping.java

```diff
@@ -0,0 +1,387 @@
+                .execute(getQueryRunner(), igniteCreateAndInsert("test_varchar"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_varchar"));
+    }
+
+    @Test
+    public void testUnboundedVarchar()
+    {
+        SqlDataTypeTest.create()
+                .addRoundTrip("varchar", "'NULL'", createUnboundedVarcharType(), "CAST('NULL' AS varchar)")
+                .addRoundTrip("varchar", "NULL", createUnboundedVarcharType(), "CAST(NULL AS varchar)")
+                .addRoundTrip("varchar", "'简体'", createUnboundedVarcharType(), "CAST('简体' AS varchar)")
+                .addRoundTrip("varchar", "'text_a'", createUnboundedVarcharType(), "CAST('text_a' AS varchar)")
+                .addRoundTrip("varchar", "'text_b'", createUnboundedVarcharType(), "CAST('text_b' AS varchar)")
+                .addRoundTrip("varchar", "'text_c'", createUnboundedVarcharType(), "CAST('text_c' AS varchar)")
+                .addRoundTrip("varchar", "'text_d'", createUnboundedVarcharType(), "CAST('text_d' AS varchar)")
+                .addRoundTrip("varchar", "'text_e'", createUnboundedVarcharType(), "CAST('text_e' AS varchar)")
+                .addRoundTrip("varchar", "'text_f'", createUnboundedVarcharType(), "CAST('text_f' AS varchar)")
+                .addRoundTrip("varchar", "'攻殻機動隊'", createUnboundedVarcharType(), "CAST('攻殻機動隊' AS varchar)")
+                .addRoundTrip("varchar", "'攻殻機動隊'", createUnboundedVarcharType(), "CAST('攻殻機動隊' AS varchar)")
+                .addRoundTrip("varchar", "'攻殻機動隊'", createUnboundedVarcharType(), "CAST('攻殻機動隊' AS varchar)")
+                .addRoundTrip("varchar", "'😂'", createUnboundedVarcharType(), "CAST('😂' AS varchar)")
+                .addRoundTrip("varchar", "'Ну, погоди!'", createUnboundedVarcharType(), "CAST('Ну, погоди!' AS varchar)")
+                .addRoundTrip("varchar", "'+-*/!234=1'", createUnboundedVarcharType(), "CAST('+-*/!234=1' AS varchar)")
+                .execute(getQueryRunner(), igniteCreateAndInsert("test_unbounded_varchar"))
+                .execute(getQueryRunner(), trinoCreateAndInsert("test_unbounded_varchar"));
+    }
+
+    @DataProvider
+    public Object[][] sessionZonesDataProvider()
+    {
+        return new Object[][] {
+                {UTC},
+                {jvmZone},
+                // using two non-JVM zones so that we don't need to worry what Ignite system zone is
+                {vilnius},
+                {kathmandu},
+                {TestingSession.DEFAULT_TIME_ZONE_KEY.getZoneId()},
+        };
+    }
+
+    private DataSetup trinoCreateAndInsert(String tableNamePrefix)
+    {
+        return new CreateAndInsertDataSetup(new TrinoSqlExecutor(getQueryRunner(), createSession()), tableNamePrefix);
+    }
+
+    private DataSetup trinoCreateAndInsert(Session session, String tableNamePrefix)
+    {
+        return new CreateAndInsertDataSetup(new TrinoSqlExecutor(getQueryRunner(), session), tableNamePrefix);
+    }
```

Comment:
Follow-up is fine. We should add `trinoCreateAsSelect`.

Reply:
https://github.com/trinodb/trino/issues/16214 filed